.github/workflows/R-CMD-check.yaml

```diff
@@ -16,30 +22,33 @@ jobs:
       fail-fast: false
       matrix:
         config:
-          - {os: macOS-latest,   r: 'release'}
+          - {os: macos-latest,   r: 'release'}
           - {os: windows-latest, r: 'release'}
           - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
           - {os: ubuntu-latest,   r: 'release'}
           - {os: ubuntu-latest,   r: 'oldrel-1'}
+          - {os: ubuntu-latest,   r: 'oldrel-2'}
+          - {os: ubuntu-latest,   r: 'oldrel-3'}
+          - {os: ubuntu-latest,   r: 'oldrel-4'}
 
     env:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
-      R_KEEP_PKG_SOURCE: true
+      R_KEEP_PKG_SOURCE: yes
 
     steps:
-      - uses: actions/checkout@HEAD
+      - uses: actions/checkout@v3
```

Comment:
I think I started using `HEAD` simply because I didn't want to update the tag name once in a while. For popular actions, I guess the HEAD branch should be fine. Anyway, since you are willing to do the upkeep work, I'm totally fine with using a specific tag :)

Reply:
Using HEAD will use the dev version of an action - using a specific tag will use a released version. I would prefer our workflow not to break because dev version of another GHA action is breaking. 

Otherwise, using `@HEAD` or  `@main` is indeed a good way to never update the workflow :)

Reply:
> I would prefer our workflow not to break because dev version of another GHA action is breaking.

My guess is that this shouldn't happen frequently. Popular actions should have tests.

The decision boils down to whether we want to spend time on updating the tags (because older tags are mostly likely to fail after some time), or spend time on dealing with potential bugs in dev versions of actions. There isn't a clear answer.

This also reminds me of the Travis CI time. Back then we didn't use tags and were essentially using the dev version every time. Bugs did occur, but were still relatively rare.

Reply:
Probably yes. I don't know really how actions are tested... 

We are using `v3` tag so we'll get update for all new version they release 3.x.z until they do a v4. 

I'll prefer to keep it that way, and I'll to the update each year. Until I don't and i'll put back HEAD maybe... 😅